### PR TITLE
10.0 fix access restricted do nothing with groups

### PR DESCRIPTION
**Pull request by iledarn**

### CODE REVIEW

File: access_restricted/models/res_users.py [hunk 1]
```diff
@@ -52,24 +52,25 @@ def get_application_groups(self, domain=None):
...
-            # `isinsnance` check is a non-xmplrpc proof.
-            # allow to remove users from a group when a user uncheck group_XXX field in settings
-            # ``all(u[0] == 3 for u in users)`` is to be sure that all operations are for removing.
-            # `(3, id)` tuple removes the record from the set (the Many2many field `users`)
-            self = self.sudo()
-        if config and self.env['res.users'].has_group('access_restricted.group_allow_add_implied_from_settings'):
-            classified = config._get_classified_fields()
+        config = self.env.context.get('config')
+
+        # `isinsnance` check is a non-xmplrpc proof.
```
Comment (yelizariev) [inline access_restricted/models/res_users.py]: isinsnance

File: access_restricted/models/res_users.py [hunk 2]
```diff
@@ -52,24 +52,25 @@ def get_application_groups(self, domain=None):
...
-                allowed_implied = [group[2].id for group in classified['group']]
-            if implied_ids and implied_ids[0][1] in allowed_implied:
-                # allow to add implied groups from `res.config.settings`
+            classified_group = config._get_classified_fields()['group']
+            # when `res.config.settings`'s `execute` method writes the `users` field to group,
+            # it is always to remove users and the `users` field is the only key in the write dict
+            users = vals.get('users')
+            if users and len(vals) == 1 and all(u[0] == 3 for u in users) or \
+               implied_ids and implied_ids[0][1] in [group[2].id for group in classified_group] and \
+               self.env['res.users'].has_group('access_restricted.group_allow_add_implied_from_settings'):
```
Comment (yelizariev) [inline access_restricted/models/res_users.py]: The condition is too difficutl to read

File: access_restricted/doc/changelog.rst [hunk 3]
```diff
@@ -1,6 +1,11 @@
 Updates
 =======
 
+`1.3.3`
+-------
+
+- **Fix:** when edit settings - only allow to exclude users from groups if there is no permission. Allow add implied groups only if a user in a special group
```
Comment (yelizariev) [inline access_restricted/doc/changelog.rst]: It's not obvious what is the difference with previous update

File: access_restricted/models/res_users.py (hunk 1 above)
Comment (iledarn) [reply access_restricted/models/res_users.py]: @yelizariev , done

**Pull request merged**